src/Umbraco.Core/Models/ImageUrlGenerationOptions.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace Umbraco.Core.Models
+{
+    public class ImageUrlGenerationOptions
+    {
+        public string ImageUrl { get; set; }
+        public int? Width { get; set; }
+        public int? Height { get; set; }
+        public decimal? WidthRatio { get; set; }
+        public decimal? HeightRatio { get; set; }
+        public int? Quality { get; set; }
+        public string ImageCropMode { get; set; }
```

Comment:
Think it's worth making this into an `enum`?  Just as it has a fixed set of valid values.  Not sure though, as I guess we want this class to be generic for image processor and any other potential implementations.

Reply:
Yeah, on this one I think we probably shouldn't go with an Enum, just because the exact values that are allowed for any of the three enums are processor dependent, i.e. ImageCropMode, ImageCropAnchor, and AnimationProcessMode.